### PR TITLE
Init API load image

### PR DESCRIPTION
**Pull request by NguyenDinhTrong99**
## [Load image from URL]
 - Create class load image from URL use Aysnc Task
 - Add method static in class Common
### [Redmine link]
 - Create API get Image:
https://edu-redmine.sun-asterisk.vn/issues/17789

### CODE REVIEW

File: app/src/main/java/com/trongdeptrai/soundcloud/screen/home/HomeFragment.java [hunk 1]
```diff
@@ -66,6 +68,7 @@ public void initListener() {
 
     @Override
     public void onGetTrendingTrackSucceed(List<Track> data) {
+        Log.d(TAG, "SIZE:\t " + data.size());
```
Comment (thaohq-0814) [inline app/src/main/java/com/trongdeptrai/soundcloud/screen/home/HomeFragment.java]: remove

File: app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/GenresAdapter.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        mOnItemRecyclerViewClickListener.onItemClickListener(list, item);
+    }
+
+    static class TrendingItem extends RecyclerView.ViewHolder implements View.OnClickListener {
+        private Context mContext;
+        private List<Track> mTracks;
+        private RecyclerView mRecyclerViewTrending;
+        private OnItemRecyclerViewClickListener<List<Track>, Track> mListener;
+
+        TrendingItem(Context context, List<Track> tracks, @NonNull View itemView,
```
Comment (thaohq-0814) [inline app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/GenresAdapter.java]: remove context

File: app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/GenresAdapter.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            OnItemRecyclerViewClickListener<List<Track>, Track> listener) {
+        mOnItemRecyclerViewClickListener = listener;
+    }
+
+    @Override
+    public void onItemClickListener(List<Track> list, Track item) {
+        mOnItemRecyclerViewClickListener.onItemClickListener(list, item);
+    }
+
+    static class TrendingItem extends RecyclerView.ViewHolder implements View.OnClickListener {
```
Comment (thaohq-0814) [inline app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/GenresAdapter.java]:
```suggestion
    static class TrendingItemViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
```

File: app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/GenresAdapter.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            adapter.setOnItemRecyclerViewClickListener(mListener);
+        }
+
+        @Override
+        public void onClick(View view) {
+            mListener.onItemClickListener(mTracks, mTracks.get(getAdapterPosition()));
+        }
+    }
+
+    static class GenreItem extends RecyclerView.ViewHolder {
```
Comment (thaohq-0814) [inline app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/GenresAdapter.java]:
```suggestion
    static class GenreItemViewHolder extends RecyclerView.ViewHolder {
```

File: app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/GenresAdapter.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        }
+
+        void bindView() {
+            TrendingTrackAdapter adapter = new TrendingTrackAdapter(mContext, mTracks);
+            mRecyclerViewTrending.setAdapter(adapter);
+            adapter.setOnItemRecyclerViewClickListener(mListener);
+        }
+
+        @Override
+        public void onClick(View view) {
```
Comment (thaohq-0814) [inline app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/GenresAdapter.java]: remove

File: app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+    void setOnItemRecyclerViewClickListener(
+            OnItemRecyclerViewClickListener<List<Track>, Track> listener) {
+        mOnItemRecyclerViewClickListener = listener;
+    }
+
+    static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        private Context mContext;
+        private List<Track> mTracks;
+        private TextView txtName;
```
Comment (thaohq-0814) [inline app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java]: rename

File: app/src/main/res/layout/item_genres.xml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    app:cardCornerRadius="@dimen/dp_4"
+    app:cardElevation="@dimen/dp_4"
+    >
+     <RelativeLayout
+         android:layout_width="match_parent"
+         android:layout_height="wrap_content">
+        <LinearLayout
+            android:id="@+id/linear_layout"
```
Comment (thaohq-0814) [inline app/src/main/res/layout/item_genres.xml]:
```suggestion
            android:id="@+id/layoutSomething"
```

File: app/src/main/res/layout/item_genres.xml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+             android:layout_marginStart="@dimen/dp_4"
+             android:layout_marginTop="@dimen/dp_4"
+             android:text="@string/title_country"
+             android:textSize="@dimen/sp_20"
+             android:textStyle="normal"
+             style="@style/TitleText"
+             />
+
+         <TextView
+             android:id="@+id/txtMore"
```
Comment (thaohq-0814) [inline app/src/main/res/layout/item_genres.xml]:
```suggestion
             android:id="@+id/textViewMore"
```

**Review by thaohq-0814 — APPROVED** [review]

File: app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+    void setOnItemRecyclerViewClickListener(
+            OnItemRecyclerViewClickListener<List<Track>, Track> listener) {
+        mOnItemRecyclerViewClickListener = listener;
+    }
+
+    static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        private Context mContext;
+        private List<Track> mTracks;
+        private TextView textViewName;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java]:
```suggestion
        private TextView mTextViewName;
```

File: app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+        private TextView textViewSinger;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java]:
```suggestion
        private TextView mTextViewSinger;
```

File: app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+        private ImageView imageViewSong;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java]:
```suggestion
        private ImageView mImageViewSong;
```

File: app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+            textViewSinger = itemView.findViewById(R.id.textViewNameSingerTrendItem);
+            imageViewSong = itemView.findViewById(R.id.imageViewSongTrendItem);
+            itemView.setOnClickListener(this);
+        }
+
+        void bindView() {
+            Track track = mTracks.get(getAdapterPosition());
+            textViewName.setText(track.getTitle());
+            textViewSinger.setText(track.getArtistName());
+            Glide.with(mContext)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java]: dùng bitmap load ảnh e. chưa được dùng lib

File: app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+                    .load(getBigImageUrl(track.getArtworkUrl()))
+                    .error(R.drawable.ic_music_default)
+                    .into(imageViewSong);
+        }
+
+        @Override
+        public void onClick(View view) {
+            mListener.onItemClickListener(mTracks, mTracks.get(getAdapterPosition()));
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/trongdeptrai/soundcloud/screen/home/adapter/TrendingTrackAdapter.java]: check null mListener

File: app/src/main/res/layout/item_genres.xml [hunk 14]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="utf-8"?>
+<com.google.android.material.card.MaterialCardView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/item_genres.xml]:
```suggestion
<com.google.android.material.card.MaterialCardView 
    xmlns:android="http://schemas.android.com/apk/res/android"
```

File: app/src/main/res/layout/item_sub_trending.xml [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+        <TextView
+            android:id="@+id/textViewNameSingerTrendItem"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_margin="@dimen/dp_4"
+            android:textColor="@android:color/white"
+            android:textSize="@dimen/sp_18"
+            />
+    </LinearLayout>
+
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/item_sub_trending.xml]: remove blank

File: app/src/main/res/layout/item_trending.xml [hunk 16]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.recyclerview.widget.RecyclerView
+    android:id="@+id/recyclerViewTrending"
+    xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/item_trending.xml]: reformat code

**Pull request merged**